chesster.js

```diff
@@ -27,177 +30,161 @@ var BOARD_5_RATING = 16;
 var BOARD_6_NAME = 17;
 var BOARD_6_RATING = 19;
 
-var LONEWOLF_CHANNEL_NAMES = [
-    "lonewolf-general",
-    "lonewolf-gamelinks",
-    "lonewolf-schedule",
-    "lonewolf-results",
-    "unstable_bot-lonewolf"
-];
-var TEAM_CHANNEL_NAMES = [
-    "general",
-    "team-general",
-    "team-gamelinks",
-    "team-schedule",
-    "team-results",
-    "unstable_bot"
-];
-var LONEWOLF_TARGETS = LONEWOLF_CHANNEL_NAMES.concat([
-    "lonewolf",
-    "lone",
-    "wolf",
-    "lw",
-    "30",
-    "3030",
-]);
-var TEAM_TARGETS = TEAM_CHANNEL_NAMES.concat([
-    "team",
-    "45",
-    "4545",
-]);
-
-if (!process.env.token) {
-  console.log('Error: Specify token in environment');
-  process.exit(1);
+/* exception handling */
+/* later this will move it its own module */
+
+function exception_handler(todo, on_error){
+    try{
+       todo();
+    }catch(e){
+        var error_log = "An error occurred:" +
+            "\nDatetime: " + new Date() +
+            "\nError: " + JSON.stringify(e) +
+            "\nStack: " + e.stack;
+        console.log(error_log);
+        fs.appendFile("./errors_log", error_log, function(err) {
+            if(err) {
+                console.log("failed to write to the file...")
+                console.log(e);
+                return console.log(err);
+            }
+        });
+        on_error && on_error();
+        console.log("\n");
+    }
+}
+
+function bot_exception_handler(bot, message, todo){
+    exception_handler(todo, function(){
+        console.log("Message: " + JSON.stringify(message));
+        bot.reply(message, "Something has gone terribly terribly wrong. Please forgive me.");
+    });
+}
+
+function critical_path(todo){
+    exception_handler(todo, function(){
+        console.log("An exception was caught in a critical code-path. I am going down.");
+        process.exit(1);
+    });
+}
+
+/* static entry point */
+
+var config;
+
+exception_handler(function(){ 
+    var config_file = process.argv[2] || "config.json";
+
+    console.log("Loading configuration from " + config_file);
+    config = JSON.parse(fs.readFileSync(config_file, 'utf8'));
```

Comment:
the encoding is technically utf-8 (with the dash). I think most contexts will happily accept the non-dash version, but some don't.